### ISSUE TITLE
User Metadata string:to_lower fails

### ISSUE THREAD

**Issue by drewkerrigan** [basho/yokozuna#183]
From calling this in riak-erlang-client:

```
{ok, Pid} = riakc_pb_socket:start_link("127.0.0.1", 8087).
Object = riakc_obj:new(<<"test">>, <<"usermeta">>, <<"data">>).
MD1 = riakc_obj:get_update_metadata(Object).
MD2 = riakc_obj:set_user_metadata_entry(MD1,{<<"Key1">>,<<"Value1">>}).
Object2 = riakc_obj:update_metadata(Object,MD2).
riakc_pb_socket:put(Pid, Object2).
```

I actually triggered this from elixir, above is what I was doing translated to Erlang

Stack:

```
Reason:     {function_clause,[{string,to_lower,[<<"my_keya">>],[{file,"string.erl"},{line,493}]},{yz_doc,'-get_user_meta/1-lc$^0/1-0-',1,[{file,"src/yz_doc.erl"},{line,141}]},{yz_doc,get_user_meta,1,[{file,"src/yz_doc.erl"},{line,141}]},{yz_doc,extract_tags,1,[{file,"src/yz_doc.erl"},{line,120}]},{yz_doc,make_doc,6,[{file,"src/yz_doc.erl"},{line,68}]},{yz_doc,'-make_docs/5-lc$^0/1-0-',6,[{file,"src/yz_doc.erl"},{line,52}]},{yz_kv,index,8,[{file,"src/yz_kv.erl"},{line,159}]},{riak_kv_vnode,perform_put,3,[{file,"src/riak_kv_vnode.erl"},{line,1092}]}]}
     Offender:   [{pid,<0.929.0>},{name,undefined},{mfargs,{riak_core_vnode,start_link,undefined}},{restart_type,temporary},{shutdown,300000},{child_type,worker}]
```

**Comment by coderoshi** [comment]
Strange, I thought I fixed this. Thanks for the heads up.

Also... Elixir?

**Comment by rzezeski** [comment]
This was fixed by #223.
